### PR TITLE
azurerm_vpn_site - support new property o365_policy

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new property `o365_policy`.

--- PASS: TestAccVpnSite_basic (301.95s)
--- PASS: TestAccVpnSite_complete (303.11s)
--- PASS: TestAccVpnSite_requiresImport (338.98s)
--- PASS: TestAccVpnSite_o365Policy (423.74s)
--- PASS: TestAccVpnSite_update (578.85s)

image

### CODE REVIEW

File: website/docs/r/vpn_site.html.markdown [hunk 1]
```diff
@@ -96,6 +98,22 @@ A `link` block supports the following:
...
+---
+
+A `breakout_category` block supports the following:
+
+* `allow_category_enabled` - (Optional) Is allow category controlled? Defaults to `false`.
+
+* `default_category_enabled` - (Optional) Is default category controlled? Defaults to `false`.
+
+* `optimize_category_enabled` - (Optional) Is optimize category controlled? Defaults to `false`.
+
```
Comment (katbyte) [inline website/docs/r/vpn_site.html.markdown:116]: i think this needs to be reworked as its not very intuitive what is being set.

what does it mean by breakout category? what is allow vs default vs optimise? should it be o365_breakout_policy? or o365_breakout_catagory_policy

allow_controlled = true etc?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/vpn_site.html.markdown (hunk 1 above)
Comment (katbyte) [reply website/docs/r/vpn_site.html.markdown:116]: @neil-yechenwei

Comment (neil-yechenwei) [reply website/docs/r/vpn_site.html.markdown]: @katbyte , I think service team would expand the sub properties of `o365_policy` to introduce other o365 policies in the future. So suggest keep `o365_policy`. I updated `breakout_category` to `traffic_category`. See more details from https://docs.microsoft.com/en-us/microsoft-365/enterprise/microsoft-365-vpn-split-tunnel?view=o365-worldwide.

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please take another look. Thanks in advance.

File: website/docs/r/vpn_site.html.markdown [hunk 2]
```diff
@@ -96,6 +98,22 @@ A `link` block supports the following:
...
+---
+
+A `traffic_category` block supports the following:
+
+* `allow_controlled` - (Optional) Is allow category controlled? Defaults to `false`.
+
+* `default_controlled` - (Optional) Is default category controlled? Defaults to `false`.
+
+* `optimize_controlled` - (Optional) Is optimize category controlled? Defaults to `false`.
```
Comment (katbyte) [inline website/docs/r/vpn_site.html.markdown:115]: i'm still not sure what these values are doing? could we eitehr fixuyp the property names or the descriptions?

Comment (neil-yechenwei) [reply website/docs/r/vpn_site.html.markdown:115]: [Here](https://docs.microsoft.com/en-us/microsoft-365/enterprise/urls-and-ip-address-ranges?view=o365-worldwide) are the ip ranges for office that fall within specific Allow/Optimize/Default categories of traffic. What this tells is essentially what kind of traffic from the branch should go directly to Office and what can flow via virtual wan.

[Here](https://docs.microsoft.com/en-us/microsoft-365/enterprise/microsoft-365-network-connectivity-principles?view=o365-worldwide#new-office-365-endpoint-categories) is the defintion of `Allow/Default/Optimize` endpoint.

**Comment by neil-yechenwei** [comment]
@katbyte, thanks for your comments. I've updated code. Please take another look. Thanks in advance.

**Review by katbyte — APPROVED** [review]
LGTM 🍔

**Pull request merged**